Use kickstart for Dotnet templates

rideam opened the pull request:
- kickstart file for setting up an Application to use with the Donet templates

TheMiniDriver committed:
added kickstart script

TheMiniDriver committed:
remove whitespace

rideam committed:
Merge branch 'main' into dotnet-templates

rideam committed:
add .env file for kickstart

rideam committed:
add script to test webapi

rideam committed:
add script to test webapi

mark-robustelli reviewed (changes requested):
I think there may be a bit of a misunderstanding.  It looks like you created additional Kickstarts.  Is there a reason we needed additional instead of just the one in the root? I think the documentation on the users could go there as well.  If there is a problem with that or this is not clear, please let me know.

rideam commented:
@mark-robustelli  the kickstart we added in the dotnet directory makes additional configurations which may not apply to all the example scripts in this repo thats why we added it as a separate kickstart instead of modifying the one in the root directory

mark-robustelli commented:
@rideam OK that is understandable, but I think it makes sense to add the things needed to the original kickstart.  It looks pretty basic and don't think it would be a problem.  I will let @mooreds chime in for the final OK on that though.

mooreds commented:
I don't understand the issue, @mark-robustelli . Can you explain it to me, please?

mark-robustelli commented:
@mooreds, there is a kickstart in the root of this project.  @rideam creates another kickstart in the dotnet folder.  I think there should be only one kickstart in the root and @rideam should update the one, not add a second.  Not necessarily replace it with the one he created but add the application and other items using the existing.  I spun up the existing and it looks pretty basic, I think it will be easy enough to add to.

mooreds commented:
@mark-robustelli if this kickstart is part of the .NET templates, why is it not included in that project? Is there some reason we can't do that?

mark-robustelli commented:
That's why I ran it by you first.  That seems like the thing to do.  @rideam Can you move it to the project?   That makes the most sense.

rideam committed:
consolidate kickstart

rideam commented:
@mark-robustelli I have moved the kickstart changes to the main kickstart

mark-robustelli commented on dotnet/templates/Program.cs:
```diff
@@ -0,0 +1,80 @@
+namespace TestWebAPI
```
I think this should have been moved to the Template repo as well right?

mark-robustelli commented:
There are still a couple of files associated with this pull request.  Shouldn't they be in the template repo as well?  And then I think we can close this request right?

sixhobbits commented:
@mark-robustelli - just jumping in here to try understand the use cases a bit better. Our understanding is that

- We have one repository (https://github.com/ritza-co/fusionauth-dotnet-templates) that is primarily the "nuget package" - these are the dotnet templates that will be published to nuget and can be used by anyone (potentially without further help, examples, instructions etc).

- Then we have `fusionauth-example-client-libraries` - this repository - that lets the user get started with those templates (or with other languages and frameworks) more easily by also including a kickstart, docker-compose, etc. If a user wants to create an app from the templates and use Fusionauth for the first time, then this will help them, but some users might be only interested in the templates themselves (or in using the other client libraries directly without these examples)

Then we have the docs changes (PR to fusionauth-site), which show how to install from nuget (after the other repo has been published), and link to the more detailed README in this repository for those users who are setting Fusionauth up for the first time.

It is possible to restructure everything and publish the docker and kickstart stuff on nuget as well, but we think having them separate and keeping the examples in this repository makes more sense, so just wanted to double check before we go ahead and do the restructure.

mark-robustelli commented:
@sixhobbits Things are becoming a little more clear.  Thanks for the explanation.  I like the way you broke it down so will try to respond in the same way.

- .Net Template Repo (https://github.com/ritza-co/fusionauth-dotnet-templates) - I think we are good here and in agreement.
- fusionauth-example-client-libraries - (the PR we are currently on) - this is where I think we can clean things up a bit.
     - I like the changes to kickstart/kickstart.json and think that is in the right place
     - I'd like to see the README.md in the root of the project updated with the kickstart info.
           - This would contain everything needed to get an instance of FusionAuth up and running for the samples. 
           - how to spin up, docker compose up -d, etc
     - Then the README.md in the dotnet/template directory would contain all the stuff specific to the templates. 
           - The application name, client secret etc.
           - how to use the templates and run them against the kickstart instance.
- fusion-site - (https://github.com/FusionAuth/fusionauth-site/pull/2385). I think this is good as long as the links are pointing to the right place when we finish here.

If we are in agreement on what should be where, please make the changes and let me know when to re-review this. If you would like to discuss further, please let me know.

rideam committed:
feedback updates

rideam committed:
feedback updates

rideam commented:
@mark-robustelli I have implemented changes according to your comment above and you can re-review

mark-robustelli reviewed (approved):
LGTM
